airflow/providers/amazon/aws/executors/ecs/README.md

```diff
@@ -0,0 +1,196 @@
+
+You can optionally mount a local directory as a volume on the container during run-time. This will allow you to make change to files on the mounted directory, and have those changes be reflected in the container. To do this, run the following command:
+
+```
+docker run --volume /abs/path/to/local/dir:/abs/path/to/remote/dir <image_name>
+```
+
+Note: Doing this will overwrite the contents of the directory on the container with the contents of the local directory.
+
+### Installing Python Dependencies
+
+This Dockerfile supports installing Python dependencies via `pip` from a `requirements.txt` file. Place your `requirements.txt` file in the same directory as the Dockerfile. If it is in a different location, it can be specified using the `requirements_path` build-argument. Keep in mind the Docker context when copying the `requirements.txt` file. Uncomment the two appropriate lines in the Dockerfile that copy the `requirements.txt` file to the container, and run `pip install` to install the dependencies on the container.
+
+### Building Image for ECS Executor
+
+Detailed instructions on how to use the Docker image, that you have created via this readme, with the ECS Executor can be found [here](link_to_how_to_guide).
+
+
+## Logging
+
+Airflow tasks executed via this executor run in ECS containers within the configured VPC. This means that logs are not directly accessible to the Airflow Webserver and when containers are stopped, after task completion, the logs would be permanently lost.
+
+Remote logging can be employed when using the ECS executor to persist your Airflow Task logs and make them viewable from the Airflow Webserver.
+
+### Configuring Remote Logging
+
+There are many ways to configure remote logging and several supported destinations. A general overview of Airflow Task logging can be found [here](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html). Instructions for configuring S3 remote logging can be found [here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/s3-task-handler.html) and Cloudwatch remote logging [here](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/cloud-watch-task-handlers.html).
+Some important things to point out for remote logging in the context of the ECS executor:
+
+ - The configuration options for Airflow remote logging must be configured on the host running the Airflow Webserver (so that it can fetch logs from the remote location) as well as within the ECS container running the Airflow Tasks (so that it can upload the logs to the remote location). See [here](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) to read more about how to set Airflow configuration via config file or environment variable exports.
+ - Adding the Airflow remote logging config to the container can be done in many ways. Some examples include, but are not limited to:
```

Comment:
Related to above, we should set the expectation that however Airflow overall is configured, is also used for these ECS containers. Suggesting bespoke config for the tasks is asking for problems eventually imo.

Reply:
I'll add, with KubernetesExecutor, it's assumed that the "deployment manager" handles this themselves in the pod_template_file. We should use a similar approach for our example here.

Reply:
I'd still prefer to keep this explicit point for point walkthrough in this section, since it was a tricky part for folks. But we can add a note somewhere that, generally, all config should be the same. I can't think of a reason off the top of my head that it should be otherwise.

Reply:
> generally, all config should be the same

Not just generally, Airflow assumes it's all consistent. This includes things like airflow_local_settings / pluggy hooks as well. And this is the tricky part when running tasks on a remote cluster. The onus is definitely on the deployment manager, but we should make sure folks are aware of that responsibility and guide them on the right approach to pull it off in a sane way.

e.g. there is a note [here](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/executor/kubernetes.html#example-pod-templates) for KE. One could argue that should be more prominent and comprehensive!

So yeah, related to above, the focal point should be that all config is consistent - logging is just one example.

Reply:
Another thought: KE tends to be a little easier because most often, you are running in k8s already and can reuse the config stuff to build a comprehensive pod_template_file (e.g. use the same secret for db creds). It doesn't sound like we have that same luxury here though.

Reply:
Just merging these threads, here's my reply in the other which I think applies here: https://github.com/apache/airflow/pull/34381#discussion_r1336275811

> Another thought: KE tends to be a little easier because most often, you are running in k8s already and can reuse the config stuff to build a comprehensive pod_template_file (e.g. use the same secret for db creds). It doesn't sound like we have that same luxury here though.

Yeah, I think the same can apply for folks running Airflow itself (scheduler, webserver, etc) on ECS, which is fairly common. And ECS has mechanisms to export a shared config to all places (you can use the same TaskDefinition for all the Tasks including worker tasks or using environemntFiles). We can make a note of this as well.

Reply:
Added updates to the language and a couple note dialogs pointing out configuration consistency